### DISCUSSION TITLE
How to run examples?

### DISCUSSION THREAD

**Discussion by imbolc** [framesurge/perseus#51] [Q&A] [answered]
I tried to use bonnie: `bonnie dev example tiny serve`, but got an error:
```
error[E0308]: mismatched types
  --> src/lib.rs:51:20
   |
51 |         locales => &get_locales()
   |                    ^^^^^^^^^^^^^^ expected struct `Locales`, found struct `perseus::locales::Locales`
   |
   = note: expected reference `&Locales`
              found reference `&perseus::locales::Locales`
   = note: perhaps two different versions of crate `perseus` are being used?
```

Then I tried to compile the dev version of the cli by hands:
```sh
ln -s examples/basic/.perseus packages/perseus-cli/.perseus
cargo install --path packages/perseus-cli
perseus -V
perseus-cli 0.3.0-beta.6
cd examples/tiny
perseus serve
```
but got the same error

**Comment by arctic-hen7** [comment]
This process is currently a little convoluted, and unfortunately not well documented (sorry!). I'll release 0.3.0-beta.7 very soon, and I'll include a `bonnie setup` script that'll prepare everything for locale development. In the meantime though, create `examples/basic/.perseus/dist/static` as an empty directory and then run `bonnie copy-subcrates`. Things should then work as expected, please let me know! For future, I'll streamline this process ASAP, thanks for bringing this to my attention.

**Reply by imbolc** [reply]
Unfortunately the same error. No need to investigate it though, I'm ok waiting for the release :)

**Reply by arctic-hen7** [reply]
Thanks! I've just released beta 7, unfortunately without these updates because I need to get the website working ASAP, but they'll be added to `main` directly after and deployed in the next release (but they don't actually need a release technically).

**Comment by arctic-hen7** [comment, marked as answer]
As of now (not a release, but the latest commit), there's a new command `bonnie setup` that will prepare your local development environment for working with Perseus! Running that should solve your problems (I'd advise starting from scratch though for good measure).

*Also, note that your manual method won't work because it's using the released version of the Perseus CLI, which is different from the local version that the example expects. When working on the repo, you're ahead of the latest release, so `bonnie dev example basic serve` uses the CLI in `packages/perseus-cli`.*

**Reply by imbolc** [reply]
Seems to be working now, thank you :)
